tests/bindings/python/test_generator_profile.py

```diff
@@ -1,1477 +1,1511 @@
 
-        p.setHasInterface(False)
-        self.assertFalse(p.hasInterface())
+        def working_name(entry):
+            name = re.sub('([A-Z]{1})', r'_\1', entry).lower()
 
-    @unittest.skip("Create tests script")
-    def test_create_tests(self):
-      import re
-      from libcellml import GeneratorProfile
-
-      p = GeneratorProfile()
-
-      seen = {}
-      def working_name(entry):
-       name = re.sub('([A-Z]{1})', r'_\1', entry).lower()
-       if name.startswith("set"):
-         name = name[4:]
-
-       return name
-
-      def ignore(entry):
-       if entry.startswith("_") or entry in ["Profile", "profile", "this", "thisown"]:
-         return True
-
-       return False
-
-      def visited(entry):
-       result = entry in seen
-
-       return entry in seen
-       if entry not in seen:
-         seen[entry] = True
-         return False
-
-       return True
-
-      def getSetNames(entry):
-        if entry.startswith("set"):
-          getter = entry[3:]
-          getter = getter[0].lower() + getter[1:]
-          return [getter, entry]
-        else:
-          setter = entry
-          setter = "set" + setter[0].upper() + setter[1:]
-          return[entry, setter]
-
-      def printStringMethod(entry):
-
-        [getter, setter] = getSetNames(entry)
-        content = getattr(p, getter)()
-        content = content.replace("\n", "\\n")
-        print("        from libcellml import GeneratorProfile")
-        print("        ")
-        print("        g = GeneratorProfile()")
-        print("        ")
-        print(f"        self.assertEqual(\"{content}\", g.{getter}())")
-        print(f"        g.{setter}(\"Some text content.\")")
-        print(f"        self.assertEqual(\"Some text content.\", g.{getter}())")
-        print("        ")
-
-      def printHasMethod(entry):
-        print("        from libcellml import GeneratorProfile")
-        print("        ")
-        print("        g = GeneratorProfile()")
-        print("        ")
-        print("        self.assertFalse(g.{0}())".format(entry))
-        print("        ")
-
-      def isStringMethod(name):
-        return name.endswith("string")
-
-      def isHasMethod(name):
-        return name.startswith("has")
-
-
-      for entry in dir(p):
-       name = working_name(entry)
-       if not ignore(entry) and not name in seen:
-         seen[name] = True
-         print("    def test_{0}(self):".format(name))
-         if isStringMethod(name):
-           printStringMethod(entry)
-         elif isHasMethod(name):
-           printHasMethod(entry)
-         else:
-           print("        # Yikes missing this method: " + entry)
+            if name.startswith('set'):
+                name = name[4:]
+
+            return name
+
+        def ignore(entry):
+            if entry.startswith('_') or entry in ['Profile', 'profile', 'this', 'thisown']:
+                return True
+
+            return False
+
+        def getSetNames(entry):
```

Comment:
Shouldn't we be using underscores instead of camelCase in Python?

Reply:
Yes, ideally, we should. However, I understand that our Python bindings are currently such that they automatically use the C++ method names. I believe @hsorby might be looking into it?

Reply:
Sorry, I don't follow.  This is defining a function that you use elsewhere ... why can you not set it to be whatever you want it to be?  What's the connection to the SWIG stuff?  Certainly where you use a library function it needs to be camel case, but I don't think that's the case here?

Reply:
Oops, my bad. I was in binding mode and didn't realise it was a function defitinion in a test. IOW, I agree with you. That code is the result of @hsorby's work on Python coverage, but I am going to change it.

Reply:
Done in commit ac7d6608.